tjpcov/covariance_cluster_counts.py

```diff
@@ -69,38 +72,53 @@ def load_from_sacc(self, sacc_file):
             sacc_file (:obj: `sacc.sacc.Sacc`): SACC file object, already
             loaded.
         """
-        # Read from SACC file relevant quantities
-        self.num_z_bins = sacc_file.metadata["nbins_cluster_redshift"]
-        self.num_richness_bins = sacc_file.metadata["nbins_cluster_richness"]
-        min_mass = sacc_file.metadata["min_mass"]
-        # survey_area = sacc_file.metadata['survey_area']
-
-        min_redshifts = [
-            sacc_file.tracers[x].metadata["z_min"]
-            for x in sacc_file.tracers
-            if x.__contains__("clusters")
-        ]
-        max_redshifts = [
-            sacc_file.tracers[x].metadata["z_max"]
-            for x in sacc_file.tracers
-            if x.__contains__("clusters")
-        ]
-        min_richness = [
-            sacc_file.tracers[x].metadata["Mproxy_min"]
-            for x in sacc_file.tracers
-            if x.__contains__("clusters")
+
+        z_tracer_type = "bin_z"
+        survey_tracer_type = "survey"
+        richness_tracer_type = "bin_richness"
+
+        survey_tracer = [
+            x
+            for x in sacc_file.tracers.values()
+            if x.tracer_type == survey_tracer_type
         ]
-        max_richness = [
-            sacc_file.tracers[x].metadata["Mproxy_max"]
-            for x in sacc_file.tracers
-            if x.__contains__("clusters")
+        if len(survey_tracer) == 0:
+            self.survey_tracer_nm = ""
+            self.survey_area = 4 * np.pi
+            print(
+                "Survey tracer not provided in sacc file.\n"
+                + "We will use the default value.",
+                flush=True,
+            )
+        else:
+            self.survey_area = survey_tracer[0].sky_area * (np.pi / 180) ** 2
+
+        # Setup redshift bins
+        z_bins = [
+            v
+            for v in sacc_file.tracers.values()
+            if v.tracer_type == z_tracer_type
         ]
+        self.num_z_bins = len(z_bins)
+        self.z_min = z_bins[0].lower
+        self.z_max = z_bins[-1].upper
+        self.z_bins = np.round(
+            np.linspace(self.z_min, self.z_max, self.num_z_bins + 1), 2
+        )
+        self.z_bin_spacing = (self.z_max - self.z_min) / self.num_z_bins
```

Comment:
I am following the implementation from Nelson for now.  Once we have benchmarking / validation for the covariance, then I think it would make sense and go back to generalize.